### PR TITLE
CON-73 Send notification to admins when a room is over booked

### PR DESCRIPTION
**Pull request by codinger41**
#### Description
This Pull request allows an administrator for the converge web app to receive notifications whenever a room has been overbooked.

#### How has this been tested
- [x] Unit tests
- [x] New feature (non-breaking change which adds functionality
- [x] Integration tests

#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/mrm_api.git`
2. Setup project according to Readme.md
3. checkout to branch `story/CON-73-overbooked-rooms-notification`
4. run the following query to create an event. Make sure the eventId and other details are valid:

#### NB
- make sure the calendarId is correct.
- make sure the numberOfParticipants is more than the room capacity.

```
mutation {
  eventCheckin(
    eventId: "er0vj3v",
    calendarId: "calender id",
    numberOfParticipants: 100,
    eventTitle: "First LFA On-Boarding",
    startTime: "2018-08-14T06:00:00-07:00",
    endTime: "2018-08-14T07:00:00-07:00"
  ){
    event {
      id
      eventTitle
      room {
        id
        name
      }
    }
  }
}
```

5. To test the sockets, create an HTML file `index.html` and fill it with the following HTML code and change the location in the channel string to whatever location the cancelled event is happening, ie from `notificatons-lagos` to `notifications-kampala`

```
<!DOCTYPE html>
  <html lang="en">
  <head>
    <title>Flask_Chat_App</title>
    <script
      src="https://code.jquery.com/jquery-3.4.1.min.js"
      integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo="
      crossorigin="anonymous"></script>
  </head>
  <body>

    <h3 id="#messages" style='color: #ccc;font-size: 30px;'>No message yet..</h3>
    <div class="message_holder"></div>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/1.7.3/socket.io.min.js"></script>
    <script type="text/javascript">
      $(document).ready(function() {
        var socket = io.connect('http://127.0.0.1:5000');
        socket.on('connect', function() {
          socket.send('User has connected!');
        });
        socket.on('notifications-lagos', function(msg) {
          console.log(msg)
          console.log('Received message');
        });
      });
    </script>
  </body>
  </html>
```

6. After creating this HTML file, open it in the browser and open the console,  fire the query to cancel another event and you'd see the event object in the console of the HTML page.

#### Checklist
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] Existing unit tests pass locally with my changes
- [x] Implementation works according to expectations

#### JIRA
[CON-72](https://andela-apprenticeship.atlassian.net/jira/software/projects/CON/boards/15/backlog?label=backend&selectedIssue=CON-73)

### CODE REVIEW

File: admin_notifications/schema.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+class Query(graphene.ObjectType):
+    all_unread_notifications = graphene.Field(
+        NotificationsList,
+        description="Returns a list of admin notifications"
+    )
+
+    @Auth.user_roles('Admin')
+    def resolve_all_unread_notifications(self, info):
```
Comment (IssaIan) [inline admin_notifications/schema.py]: @codinger41, I see that the admin can only view unread notifications from the implementation here. Would it not be a good idea to enable the admin user to view all the notifications, regardless of the read status?
Also, please add the `Super_Admin` role as the super admin should also be granted these privileges.

File: admin_notifications/schema.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+class UpdateNotificationStatus(graphene.Mutation):
+    """
+        Class to update the status of a notification
+    """
+
+    class Arguments:
+        notification_id = graphene.Int(required=True)
+    notification = graphene.Field(AdminNotifications)
+
+    @Auth.user_roles('Admin')
```
Comment (IssaIan) [inline admin_notifications/schema.py]: I also believe the super admin should be granted privileges here.

File: admin_notifications/schema.py (hunk 1 above)
Comment (codinger41) [reply admin_notifications/schema.py]: @issaian I don't think the admins need to see notifications that have already been read. I'd add the super admin role.

Comment (codinger41) [reply admin_notifications/schema.py]: Why do you think admins need to see already read notifications?

Comment (IssaIan) [reply admin_notifications/schema.py]: @codinger41 the implementation without the viewing of read notifications works fine. I thought that there was a mutation to enable the admin to delete notifications. As there is none, there is no need of it then.

**Comment by rajeman** [comment]
Tasks related to admin notifications have been suspended.

**Pull request closed without merging**